#164139645 Implement user email login

chingsley opened the pull request:
#### What does this PR do?
It consumes the email login endpoint using react/redux

#### Description of Task to be completed?
Users should be able to log in to Authors Haven by providing valid email and password and having a valid token delivered to them with which they can access other routes.

#### How should this be manually tested?
- Visit the app URL `https://vidar-ah-frontend-staging.herokuapp.com/login`
- Enter your email and password
- If the credentials are valid, you'll be successfully signed in to the app
- If the credentials are invalid, you'll get error(s) messages.

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139645

#### Screenshots (if appropriate)


#### Questions:
N/A

youngestdj committed:
chore(components): create react components

- add react components

[Finishes #165211055]

Commit:
feat(user-login): design the ui with sass

- get the elements rendered on the page
- style the use using scss
- consume the endpoint for user login via email and password

[Fininshes #164139645)]

Commit:
feat(user-login): design the ui with sass

- cosume the login endpoint using react and redux
- write tests for the feature

[Fininshes #164139645)]

Commit:
--no-edit

9jaswag commented:
@chingsley why is this not ready for me to review?

Dubby20 commented on src/__tests__/views/LoginPage.test.js:
```diff
@@ -0,0 +1,65 @@
+const { mountWrapper } = getLoginPage();
+
+describe('Test the Login Page', () => {
+  it('it render the longin page without failing', () => {
```
correct the spelling error

Dubby20 replied:
correct the spelling error

9jaswag commented on package.json:
```diff
@@ -37,8 +39,11 @@
     "autoprefixer": "^9.5.0",
     "axios": "^0.18.0",
     "bootstrap": "^4.3.1",
+    "enzyme-to-json": "^3.3.5",
```
@chingsley what does this do?

chingsley replied:
It's used for `snapshot` testing.

9jaswag commented on src/assets/scss/default.scss:
```diff
@@ -0,0 +1,76 @@
+@import './settings';
```
@chingsley this is a valid Hound error....fix them and other valid ones

9jaswag commented on src/assets/scss/default.scss:
```diff
@@ -0,0 +1,75 @@
+.purple-gradient-bg {
+  background-attachment: fixed;
+  background-image: linear-gradient(to right, $deep-purple-grad, $light-purple-grad),
+  url('https://res.cloudinary.com/jessam/image/upload/v1554725326/tachina-lee-42980-unsplash_1.jpg');
```
@Dubby20 and @leksyib you guys reviewed and approved this PR visible valid Hound errors.

chingsley replied:
The use of URL is the team convention. Every code in all branch uses the same URL. Taking it out will affect other people's branches. 
I have fixed the `double-quoted strings` comment.

9jaswag reviewed (changes requested):
Fix the valid hound error and other feedback and let me see a screenshare

9jaswag commented:
@chingsley merge conflicts here to be fixed

chingsley commented on src/views/LoginPage/login.scss:
```diff
@@ -0,0 +1,56 @@
+      font-size: 1em;
+      text-decoration: none;
+
+      &:hover {
```
![image](https://user-images.githubusercontent.com/38833766/56212014-186e2580-6051-11e9-8b7f-531a76c22e29.png)

chingsley commented on src/components/loader/loader.scss:
```diff
@@ -1,6 +1,8 @@
+$faded-black: rgba(0, 0, 0, .8);
```
This contradicts the sass rule we have. Running lint prompts that we get rid of leading zeros. It sass-lint expects that we use .8 instead of 0.8. And using 0.8 will prevent commit because of the implementation of pre-commit hooks.

chingsley commented on src/components/loader/Loader.scss:
```diff
@@ -0,0 +1,18 @@
+$faded-black: rgba(0, 0, 0, .8);
```
https://github.com/andela/vidar-ah-frontend/pull/14#discussion_r275787839

chingsley commented on src/assets/scss/settings.scss:
```diff
@@ -0,0 +1,42 @@
+$color-ccc: #ccc;
+$white: #fff;
+$primary-color: #999;
+$box-shadow-color: 0 6px 20px 1px rgba(0, 0, 0, .15);
```
https://github.com/andela/vidar-ah-frontend/pull/14#discussion_r275787839

chingsley commented on src/assets/scss/settings.scss:
```diff
@@ -0,0 +1,42 @@
+$light-purple-grad: rgba(173, 83, 137, .9);
+$aaa-border: #aaa;
+$ddd-border: #ddd;
+$black-trans-bg: rgba(0, 0, 0, .6);
```
https://github.com/andela/vidar-ah-frontend/pull/14#discussion_r275787839

chingsley commented on src/assets/scss/settings.scss:
```diff
@@ -0,0 +1,42 @@
+$deep-purple: hsl(279, 68%, 19%);
+$deep-purple-grad: rgba(60, 16, 83, .9);
+$light-blue: #0575e6;
+$light-purple-grad: rgba(173, 83, 137, .9);
```
https://github.com/andela/vidar-ah-frontend/pull/14#discussion_r275787839

chingsley commented on src/assets/scss/settings.scss:
```diff
@@ -0,0 +1,42 @@
+$text-color: #6f42c1;
+$primary-color: #999;
+$deep-purple: hsl(279, 68%, 19%);
+$deep-purple-grad: rgba(60, 16, 83, .9);
```
https://github.com/andela/vidar-ah-frontend/pull/14#discussion_r275787839

chingsley commented on src/assets/scss/default.scss:
```diff
@@ -0,0 +1,75 @@
+  }
+
+  & .social-media-icons {
+    margin: 0 .8rem;
```
https://github.com/andela/vidar-ah-frontend/pull/14#discussion_r275787839

chingsley commented on src/assets/scss/default.scss:
```diff
@@ -0,0 +1,75 @@
+    border-radius: 0;
+    box-sizing: border-box;
+    color: $ddd-border;
+    font-size: .9em;
```
https://github.com/andela/vidar-ah-frontend/pull/14#discussion_r275787839

chingsley commented on src/assets/scss/default.scss:
```diff
@@ -0,0 +1,75 @@
+
+a,
+.yellow-button {
+  transition: ease-in-out .4s;
```
https://github.com/andela/vidar-ah-frontend/pull/14#discussion_r275787839

chingsley commented on src/views/LoginPage/index.jsx:
```diff
@@ -0,0 +1,127 @@
+import { SocialIcon } from 'react-social-icons';
+import {
+  Container, Row, Col, Form, Alert
+} from 'react-bootstrap';
```
![image](https://user-images.githubusercontent.com/38833766/56214690-45710700-6056-11e9-9d3b-ce934ddec5aa.png)

chingsley replied:
This contradicts the eslint rule we have configured.

chingsley commented on src/views/LoginPage/index.jsx:
```diff
@@ -0,0 +1,127 @@
+  }
+
+  return (
+    <>
```
For elements used in JSX files, we use double quotes.

chingsley commented on src/__tests__/views/LoginPage.test.js:
```diff
@@ -0,0 +1,65 @@
+        <Login {...props} />
+      </MemoryRouter>
+    </Provider>
+  );
```
This is JSX. Newline is necessary

Commit:
feat(user-login): implement user signup feature

- Resolve merge conflict to include password reset

[Delivers #164139645]

Commit:
feat(user-login): implement user signup feature

- implement user login
- fix navigation to work correctly
- make the banner.spec.js tests pass

[Delivers #164139645]

olajide1234 commented on src/__tests__/views/LoginPage.test.js:
```diff
@@ -0,0 +1,65 @@
+
+const { mountWrapper } = getLoginPage();
+
+describe('Test the Login Page', () => {
```
As discussed, please confirm the description.

olajide1234 commented on src/assets/default.scss:
```diff
@@ -50,11 +63,21 @@ h4 {
+h5 {
+  font-family: $font-family-header;
+  font-size: 0.5rem;
+}
```
I think this is a valid comment

9jaswag commented on src/redux/actions/fetching.js:
```diff
@@ -0,0 +1,18 @@
+import { START_FETCHING, STOP_FETCHING } from './actionTypes';
+
+
+export const startFetching = () => (
```
@chingsley what does this file do?

chingsley replied:
I use it to control the loader. When I make a request to the database, I will set the state property `fetching` to true by dispatching the START_FETCHING action type. When `fetching` is true, the `loader` (the round balls will show up showing the user that an action is being performed in the background). The reverse is the case for STOP_FETCHING - to stop the loader.

9jaswag replied:
I thought this is already in use? How are others achieving it?

chingsley replied:
I created the loader. I created the loader. Others did not use loader. I just asked.

9jaswag commented on src/views/LoginPage/index.jsx:
```diff
@@ -0,0 +1,127 @@
+  }
+
+  return (
+    <>
```
@chingsley if every other element is under the `div`, there's no need for a fragment.

chingsley replied:
You are right. I have taken it out.

9jaswag commented on src/views/Article/article.scss:
```diff
@@ -0,0 +1,14 @@
+    text-align: center;
+
+  }
+}
```
@chingsley fix this EOL issue

9jaswag commented:
@chingsley @leksyib test this after its merged.
